parser.y

```diff
@@ -1338,10 +1338,10 @@ import (
 %precedence remove
 %precedence lowerThenOrder
 %precedence order
-%left join straightJoin inner cross left right full natural
 
 /* A dummy token to force the priority of TableRef production in a join. */
 %left tableRefPriority
+%left join straightJoin inner cross left right full natural
```

Comment:
would it be easier to make the JOIN operator right-associative? 🤔

Reply:
NewCrossJoin() will do the reorganization, make it left-associative.

Reply:
i mean if you use `%right` here perhaps you don't need that `NewCrossJoin()` function

Reply:
Then  `t1 right join t2 right join t3` would be `t1 right join (t2 right join t3)`, it's not correct.

Reply:
why is that incorrect? `a RIGHT JOIN b RIGHT JOIN c ON w1 ON w2` is indeed equivalent to `a RIGHT JOIN (b RIGHT JOIN c ON w1) ON w2`, 
and the same for LEFT JOIN (`a LEFT JOIN b LEFT JOIN c ON w1 ON w2` == `a LEFT JOIN (b LEFT JOIN c ON w1) ON w2`)

Reply:
Sorry, the example is wrong.
Consider `select * from t1 join t2 right join t3 on t2.a=t3.a;`, if using right-associative, we got 
`select * from t1 join (t2 right join t3 on t2.a=t3.a)`. They are not the same.
```
select * from t1 join t2 join t3;
mysql> select * from t1 join t2 right join t3 on t2.a=t3.a;
+------+------+------+
| a    | a    | a    |
+------+------+------+
| NULL | NULL |    3 |
+------+------+------+
1 row in set (0.00 sec)

mysql> select * from t1 join (t2 right join t3 on t2.a=t3.a);
+------+------+------+
| a    | a    | a    |
+------+------+------+
|    1 | NULL |    3 |
+------+------+------+
1 row in set (0.00 sec)
```